src/pages/Game.tsx

```diff
@@ -168,11 +182,96 @@ const GamePage = ({ ws, currentUser }: GameProps): JSX.Element => {
     const tile = TileFactory.createTileFromStringDef(data.tile);
     const mjGameState = gameState as MahjongGameState;
     mjGameState.getDeadPile().add(tile);
-    // add validation of whether other players want to interact
-    // for now, change turn
-    mjGameState.goToNextTurn();
+
+    const mjPlayer = mjGameState.getMjPlayer();
+
+    // Ask for interaction if player was not the one who played tile
+    if (data.connectionId !== mjPlayer.getConnectionId()) {
+      const users = mjGameState.getUsers();
+      // Change state that the opponent has played a tile.
+      const opponentIndex = users.findIndex((user) => user.getConnectionId() === data.connectionId);
+      if (opponentIndex !== -1) {
+        const opponent = users[opponentIndex] as MahjongOpponent;
+        opponent.playedTile();
+      } else {
+        console.error('Game state error. Desync is likely to happen.');
+      }
+
+      mjPlayer.setAllowInteraction(true);
+      // TODO: (NextPR) set timeout, if nothing happens, send skip
+    } else {
+      // TODO (NextPR): Display msg to player who played tile to wait while others make decision
+    }
+    mjGameState.requestRedraw();
+  };
+
+  /**
+   * For PLAYED_TILE_INTERACTION
+   * @param payload PlayedTileInteraction
+   */
+  const mjPlayedTileInteraction = (payload: unknown): void => {
+    const data = payload as PlayedTileInteractionJSON;
+    // eslint-disable-next-line object-curly-newline
+    const { playedTiles, skipInteraction, meldType, success } = data;
+    // If there was an error with processing PLAYED_TILE_INTERACTION, resend message
+    if (!success) {
+      ws?.sendMessage(OutgoingAction.PLAYED_TILE_INTERACTION, {
+        playedTiles,
+        skipInteraction,
+        meldType,
+        gameId: game.gameId,
+      });
+    }
+  };
+
+  /**
+   * For INTERACTION_SUCCESS
+   * @param payload InteractionSuccessJSON
+   */
+  const mjInteractionSuccess = (payload: unknown): void => {
+    const data = payload as InteractionSuccessJSON;
+    const mjGameState = gameState as MahjongGameState;
+
+    if (data.skipInteraction) {
+      // If everybody skips, game can proceed normally
+      mjGameState.goToNextTurn();
+
+      const userIndex = mjGameState.getCurrentTurn();
+      // If it's user's turn, player can draw tile
+      if (mjGameState.getUsers()[userIndex].getConnectionId() === mjGameState.getMjPlayer().getConnectionId()) {
+        ws?.sendMessage(OutgoingAction.DRAW_TILE, { gameId: game.gameId });
+      } else {
+        // Change state that opponent will draw a tile
+        const opponent = mjGameState.getUsers()[userIndex] as MahjongOpponent;
+        opponent.drawTile();
+      }
```

Comment:
can put `mjGameState.getUsers()[userIndex]` and `mjGameState.getMjPlayer()` into variables
```ts
      const currentMjPlayer = mjGameState.getUsers()[userIndex];
      const mjPlayer = mjGameState.getMjPlayer();
      // If it's user's turn, player can draw tile
      if (currentMjPlayer.getConnectionId() === mjPlayer.getConnectionId()) {
        ws?.sendMessage(OutgoingAction.DRAW_TILE, { gameId: game.gameId });
      } else {
        // Change state that opponent will draw a tile
        const opponent = currentMjPlayer as MahjongOpponent;
        opponent.drawTile();
      }
```

Reply:
Okies